### PR TITLE
use for resume the biggest mounted swap (bsc#1071354)

### PR DESCRIPTION
**Pull request by jreidinger**
in this PBI will be also fix for merging noresume to kernel parameters with resume https://bugzilla.opensuse.org/show_bug.cgi?id=1079537

### CODE REVIEW

File: src/lib/bootloader/generic_widgets.rb [hunk 1]
```diff
@@ -73,9 +73,10 @@ def handle
 
     def help
       _(
-        "<p><b>Boot Loader Type</b><br>\n" \
-          "To select whether to install a boot loader and which bootloader to install,\n" \
-          "use <b>Boot Loader</b>.</p>"
+        "<p><b>Boot Loader</b>\n" \
+          "specifies which bootloader to install. Can be also specified <tt>None</tt> " \
```
Comment (lslezak) [inline src/lib/bootloader/generic_widgets.rb]:
```suggestion
          "specifies which bootloader to install. Can be also set to <tt>None</tt> " \
```

File: src/lib/bootloader/generic_widgets.rb [hunk 2]
```diff
@@ -73,9 +73,10 @@ def handle
...
+          "which mean that bootloader configuration is not managed by YaST2 and also " \
```
Comment (lslezak) [inline src/lib/bootloader/generic_widgets.rb]:
```suggestion
          "which means that the boot loader configuration is not managed by YaST and also " \
```

File: src/lib/bootloader/generic_widgets.rb [hunk 3]
```diff
@@ -73,9 +73,10 @@ def handle
...
+          "kernel post install script does not update bootloader configuration."
```
Comment (lslezak) [inline src/lib/bootloader/generic_widgets.rb]:
```suggestion
          "the kernel post install script does not update the boot loader configuration."
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 4]
```diff
@@ -94,8 +94,8 @@ def label
 
     def help
       _(
-        "<p><b>Set active Flag in Partition Table for Boot Partition</b><br>\n" \
-          "To activate the partition which contains the boot loader." \
+        "<p><b>Set active Flag in Partition Table for Boot Partition</b>\n" \
+          "specify if the partition which contains the boot loader will have active flag." \
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
          "specify whether the partition containing the boot loader will have active flag." \
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 5]
```diff
@@ -272,7 +272,7 @@ def label
     end
 
     def help
-      _("Tick to enable UEFI Secure Boot\n")
+      _("<p><b>Enable Secure Boot Support</b> if checked enable UEFI Secure Boot.</p>")
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
      _("<p><b>Enable Secure Boot Support</b> if checked enables UEFI Secure Boot support.</p>")
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 6]
```diff
@@ -299,15 +299,17 @@ def label
...
-          "<p>It means measuring the integrity of the boot process,\n" \
+          "instead of regular GRUB2." \
+          "It means measuring the integrity of the boot process,\n" \
           "with the help from the hardware (a TPM, Trusted Platform Module,\n" \
-          "chip).</p>\n")
+          "chip).\n")
       if grub2.name == "grub2"
-        res += _("<p>First you need to make sure Trusted Boot is enabled in the BIOS\n" \
-          "setup (the setting may be named Security Chip, for example).</p>\n")
+        res += _(" First you need to make sure Trusted Boot is enabled in the BIOS\n" \
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]: Um, this will be probably tricky, the translators might overlook the starting space or they might consider it as a typo and remove it from translations.

Either add a `TRANSLATORS:` comment above the line or change to `res += " " + _("....")`

File: src/lib/bootloader/grub2_widgets.rb [hunk 7]
```diff
@@ -299,15 +299,17 @@ def label
...
+          "setup (the setting may be named Security Chip, for example).\n")
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]: I'd quote the name here

```suggestion
          "setup (the setting may be named \"Security Chip\", for example).\n")
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 8]
```diff
@@ -299,15 +299,17 @@ def label
...
       end
 
+      res += "<p>"
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]: Is that correct? Should it rather be a closing tag?

```suggestion
      res += "</p>"
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 9]
```diff
@@ -435,16 +437,16 @@ def store
 
     def help
       _(
-        "<p><b>Protect Boot Loader with Password</b><br>\n" \
-          "At boot time, modifying or even booting any entry will require the" \
+        "<p><b>Protect Boot Loader with Password</b>\n" \
+          "at boot time, modifying or even booting any entry will require the" \
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]: This is inconsistent consistent here, the other lines use trailing space at the end of the line.

File: src/lib/bootloader/grub2_widgets.rb [hunk 10]
```diff
@@ -475,11 +477,11 @@ def contents
     def help
       # Translators: do not translate the quoted parts like "unit"
       _(
-        "<p>When a graphical console is used it allows to use various " \
+        "<p><b>Use graphical console</b> when checked it allows to use various " \
         "display resolutions. The <tt>auto</tt> option tries to find " \
         "the best one when booting starts.</p>\n" \
-        "<p>When a serial console is used the boot output " \
-        "will be printed to a serial device like <tt>ttyS0</tt>. " \
+        "<p><b>Use serial console</b> when checked it redirect the boot output " \
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
        "<p><b>Use serial console</b> when checked it redirects the boot output " \
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 11]
```diff
@@ -475,11 +477,11 @@ def contents
...
+        "a serial device like <tt>ttyS0</tt>. " \
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
        "to a serial device like <tt>ttyS0</tt>. " \
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 12]
```diff
@@ -671,12 +673,9 @@ def label
 
     def help
       _(
-        "<p> By pressing <b>Set as Default</b> you mark the selected section as\n" \
-        "the default. When booting, the boot loader will provide a boot menu and\n" \
-        "wait for the user to select a kernel or OS to boot. If no\n" \
-        "key is pressed before the timeout, the default kernel or OS will\n" \
-        "boot. The order of the sections in the boot loader menu can be changed\n" \
-        "using the <b>Up</b> and <b>Down</b> buttons.</p>\n"
+        "<p><b>Default Boot Section</b> selects boot section to be booted as default.\n" \
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
        "<p><b>Default Boot Section</b> selects the default section for booting.\n" \
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 13]
```diff
@@ -671,12 +673,9 @@ def label
...
+        " If YaST2 does not know yet naming of sections, it will be empty and default \n" \
```
Comment (lslezak) [inline src/lib/bootloader/grub2_widgets.rb]: Hm, this is hard to understand, could you rephrase it? I do not know how to write that better... :-/

File: src/modules/BootStorage.rb [hunk 14]
```diff
@@ -107,9 +107,19 @@ def prep_partitions
...
 
-      staging.filesystems.select { |f| f.type.is?(:swap) }.each do |swap|
-        blk_device = swap.blk_devices[0]
-        ret[blk_device.name] = blk_device.size.to_i / 1024
+      mounted = Y2Storage::MountPoint.find_by_path(staging, Y2Storage::MountPoint::SWAP_PATH.to_s)
+      if mounted.empty?
+        log.info "No mounted swap found, using fallback."
+        staging.filesystems.select { |f| f.type.is?(:swap) }.each do |swap|
+          blk_device = swap.blk_devices[0]
+          ret[blk_device.name] = blk_device.size.to_i / 1024
```
Comment (lslezak) [inline src/modules/BootStorage.rb]: The storage size does not provide any simple method for unit conversions? Like `blk_device.size.to_kiB` ?

File: test/boot_storage_test.rb [hunk 15]
```diff
@@ -28,6 +28,13 @@
         "/dev/mapper/cr_swap" => 2_095_104
       )
     end
+
+    it "returns only mounted swaps if it is available" do
```
Comment (lslezak) [inline test/boot_storage_test.rb]:
```suggestion
    it "returns only mounted swaps if any is available" do
```

File: src/lib/bootloader/grub2_widgets.rb (hunk 8 above)
Comment (jreidinger) [reply src/lib/bootloader/grub2_widgets.rb]: correct. It is the last entry, so it is not visible when I am checking it.

File: src/lib/bootloader/grub2_widgets.rb (hunk 9 above)
Comment (jreidinger) [reply src/lib/bootloader/grub2_widgets.rb]: it is not visible, so does not matter much for me. But I can add it.

File: src/modules/BootStorage.rb (hunk 14 above)
Comment (jreidinger) [reply src/modules/BootStorage.rb]: Well, only in form of `Y2Storage::DiskSize.KiB(blk_device.size)` which is a bit long for me.

File: src/lib/bootloader/generic_widgets.rb [hunk 16]
```diff
@@ -73,9 +73,10 @@ def handle
 
     def help
       _(
-        "<p><b>Boot Loader Type</b><br>\n" \
-          "To select whether to install a boot loader and which bootloader to install,\n" \
-          "use <b>Boot Loader</b>.</p>"
+        "<p><b>Boot Loader</b>\n" \
+          "specifies which bootloader to install. Can be also set to <tt>None</tt> " \
```
Comment (mvidner) [inline src/lib/bootloader/generic_widgets.rb]:
```suggestion
          "specifies which boot loader to install. Can be also set to <tt>None</tt> " \
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 17]
```diff
@@ -57,8 +57,8 @@ def label
     end
 
     def help
-      _("<p><b>Timeout in Seconds</b><br>\n" \
-        "Specifies the time the bootloader will wait until the default kernel is loaded.</p>\n")
+      _("<p><b>Timeout in Seconds</b>\n" \
+        "specifies the time the bootloader will wait until the default kernel is loaded.</p>\n")
```
Comment (mvidner) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
        "specifies the time the boot loader will wait until the default kernel is loaded.</p>\n")
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 18]
```diff
@@ -94,8 +94,8 @@ def label
 
     def help
       _(
-        "<p><b>Set active Flag in Partition Table for Boot Partition</b><br>\n" \
-          "To activate the partition which contains the boot loader." \
+        "<p><b>Set active Flag in Partition Table for Boot Partition</b>\n" \
```
Comment (mvidner) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
        "<p><b>Set Active Flag in Partition Table for Boot Partition</b>\n" \
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 19]
```diff
@@ -94,8 +94,8 @@ def label
...
+          "specify whether the partition containing the boot loader will have active flag." \
```
Comment (mvidner) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
          "specifies whether the partition containing the boot loader will have the \"active\" flag." \
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 20]
```diff
@@ -299,15 +299,17 @@ def label
     def help
       # TRANSLATORS: TrustedGRUB2 is a name, don't translate it
       res = _("<p><b>Trusted Boot</b> will install TrustedGRUB2\n" \
-          "instead of regular GRUB2.</p>\n" \
-          "<p>It means measuring the integrity of the boot process,\n" \
+          "instead of regular GRUB2." \
```
Comment (mvidner) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
          "instead of regular GRUB2.\n" \
```

File: src/lib/bootloader/grub2_widgets.rb (hunk 6 above)
Comment (mvidner) [inline src/lib/bootloader/grub2_widgets.rb]: The previous string always ends with a newline, so no initial space is needed.
```suggestion
        res += _("First you need to make sure Trusted Boot is enabled in the BIOS\n" \
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 21]
```diff
@@ -671,12 +673,9 @@ def label
...
     def help
       _(
-        "<p> By pressing <b>Set as Default</b> you mark the selected section as\n" \
-        "the default. When booting, the boot loader will provide a boot menu and\n" \
-        "wait for the user to select a kernel or OS to boot. If no\n" \
-        "key is pressed before the timeout, the default kernel or OS will\n" \
-        "boot. The order of the sections in the boot loader menu can be changed\n" \
-        "using the <b>Up</b> and <b>Down</b> buttons.</p>\n"
+        "<p><b>Default Boot Section</b> selects the default section for booting.\n" \
+        " If sections are not generated yet ( e.g. during installation) \n"
```
Comment (mvidner) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
        "If sections are not generated yet (e.g. during the installation)\n"
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 22]
```diff
@@ -671,12 +673,9 @@ def label
...
+        "then box is empty and default is picked by grub2 itself."
```
Comment (mvidner) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
        "then the box is empty and the default is picked by grub2 itself.</p>\n"
```

File: src/lib/bootloader/grub2_widgets.rb [hunk 23]
```diff
@@ -819,8 +818,8 @@ def help
       textdomain "bootloader"
 
       _(
-        "<p><big><b>Disks Order</b></big><br>\n" \
-          "To specify the order of the disks according to the order in BIOS, use\n" \
+        "<p><b>Edit Disk Boot Order</b>\n" \
+          "allows to specify the order of the disks according to the order in BIOS, use\n" \
```
Comment (mvidner) [inline src/lib/bootloader/grub2_widgets.rb]:
```suggestion
          "allows to specify the order of the disks according to the order in BIOS. Use\n" \
```

File: src/lib/bootloader/grub2_widgets.rb (hunk 6 above)
Comment (lslezak) [reply src/lib/bootloader/grub2_widgets.rb]: Ah, good point!
